### PR TITLE
Fix issues

### PR DESCRIPTION
**Pull request by johannes-moegerle**
see #159 and #161

### CODE REVIEW

**Comment by hmenke** [comment]
Maybe this could also be added to the patchset. This is to (possibly?) work around the test failure that I observed on Flathub: https://buildbot.flathub.org/#/builders/35/builds/5977/steps/8/logs/stdio
```diff
diff --git a/pairinteraction_gui/worker.py b/pairinteraction_gui/worker.py
index febc284e..b99ba013 100644
--- a/pairinteraction_gui/worker.py
+++ b/pairinteraction_gui/worker.py
@@ -16,6 +16,7 @@
 # along with the pairinteraction GUI. If not, see <http://www.gnu.org/licenses/>.
 from queue import Queue
 
+from PyQt5 import sip
 from PyQt5.QtCore import QThread, pyqtSignal
 
 
@@ -29,7 +30,8 @@ class Worker(QThread):
 
     def __del__(self):
         self.exiting = True
-        self.wait()
+        if not sip.isdeleted(self):
+            self.wait()
 
     def execute(self, stdout):
         self.stdout = stdout
```

**Comment by johannes-moegerle** [comment]
We are also calling QThread.wait in pipy_thready.py (see PipyThread.terminate), as well as a few times directly in app.py.
Should this workaround also be added there?

**Comment by hmenke** [comment]
I think the culprit here is that `self.wait()` is called inside `__del__` and the Qt library has probably already freed all the memory already which leads to the crash.

**Comment by johannes-moegerle** [comment]
The flathub fix fails in Linux / ubuntu. My guess is, this is due to PyQt5 being on version 5.14 instead of 5.15 as all the other builds.

File: pairinteraction_gui/worker.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 # along with the pairinteraction GUI. If not, see <http://www.gnu.org/licenses/>.
 from queue import Queue
 
+from PyQt5 import sip
```
Comment (hmenke) [inline pairinteraction_gui/worker.py:19]:
```suggestion
# https://www.riverbankcomputing.com/static/Docs/PyQt5/incompatibilities.html#importing-the-sip-module
try:
    from PyQt5 import sip
except ImportError:
    import sip
```

**Review by seweber — APPROVED** [review]

**Review by hmenke — CHANGES_REQUESTED** [review]
Squash the fixup, then merge.

**Review by hmenke — APPROVED** [review]

**Pull request merged**